### PR TITLE
Properly support SVGs

### PR DESCRIPTION
**Pull request by graphiteisaac**
This is a pretty simple, rudimentary addition to properly support Lustre SVG elements. There will inevitably be edge cases in the SVG spec I haven't considered, but for 99% of SVGs used on the web, this should work just fine, and be a nice helpful tool for people.

I added a couple of functions which pass an "is_svg" boolean down the chain, as it's helpful to consider context for elements, and considered a couple of edge cases in the attribute printer (eg, svg can't set the width with attribute.width, viewBox is case sensitive). There is one new test which covers both these cases, proving it works :)

### CODE REVIEW

File: app/src/app.gleam [hunk 1]
```diff
@@ -19,9 +20,9 @@ pub type Model {
   Model(rendered_lustre: String, copy_button_text: String)
 }
 
-const copy_button_default_text = "Copy 🧟"
+const copy_button_default_text = "Copy"
 
-const copy_button_copied_text = "Copied 🥰"
+const copy_button_copied_text = "Copied"
```
Comment (lpil) [inline app/src/app.gleam]: Why have these been removed?

File: src/html_lustre_converter.gleam [hunk 2]
```diff
@@ -14,12 +15,19 @@ import javascript_dom_parser.{type HtmlNode, Comment, Element, Text} as parser
 /// import lustre/element.{element, text}
 /// ```
 ///
+/// If the source document contains SVGs, we need one more import:
+/// ```gleam
+/// import lustre/element/svg
+/// // or
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: or?

File: src/html_lustre_converter.gleam [hunk 3]
```diff
@@ -184,14 +295,26 @@ fn print_element(
...
     | "var"
     | "video" -> {
-      let children = wrap(print_children(children, ws), "[", "]")
+      let children = wrap(print_children(children, ws, False), "[", "]")
       doc.from_string("html." <> tag)
       |> doc.append(wrap([attributes, children], "(", ")"))
     }
 
+    "svg" -> {
+      // TODO: Fix double compute, avoid initial compute if we don't go to SVG land
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: What does this mean?

File: src/html_lustre_converter.gleam [hunk 4]
```diff
@@ -224,10 +347,16 @@ fn get_text_content(nodes: List(HtmlNode)) -> String {
 fn print_children(
   children: List(HtmlNode),
   ws: WhitespaceMode,
+  is_svg: Bool,
```
Comment (lpil) [inline src/html_lustre_converter.gleam]: Prefer a custom type to a bool 🙏

**Review by lpil — COMMENTED** [review]
Ahh thank you!! I've left a few notes

File: app/src/app.gleam (hunk 1 above)
Comment (graphiteisaac) [reply app/src/app.gleam]: I put in an SVG icon for the copy, which is perhaps a bit less joyous but is a bit more semantically clear

File: src/html_lustre_converter.gleam (hunk 3 above)
Comment (graphiteisaac) [reply src/html_lustre_converter.gleam]: Currently, we already compute the attributes at the start of the function, but then when in svg we compute them a second time with the svg-friendly output. I ended up leaving it in when I found no meaningful performance impact, I found it was not necessarily needed. Should have preferenced speaking about it in the PR over leaving a dud TODO :P

File: src/html_lustre_converter.gleam (hunk 4 above)
Comment (graphiteisaac) [reply src/html_lustre_converter.gleam]: Will update :)

File: app/src/app.gleam (hunk 1 above)
Comment (lpil) [reply app/src/app.gleam]: One change per PR please, revert this

File: src/html_lustre_converter.gleam (hunk 3 above)
Comment (lpil) [reply src/html_lustre_converter.gleam]: Can just remove the comment I think

**Comment by graphiteisaac** [comment]
Thanks very much Louis :) I've made numerous changes based on your feedback now.

**Review by lpil — APPROVED** [review]
thank you!!!!!
